motion/motionprocessor.go

```diff
@@ -159,6 +187,41 @@ func (mp *MotionProcessor) Process(rawFrame []byte) error {
 	return nil
 }
 
+func (mp *MotionProcessor) cameraTempControll(frame *cptvframe.Frame) {
+	if mp.finishedWarmup {
+		if frame.Status.TimeOn-frame.Status.LastFFCTime > 2*time.Minute &&
+			frame.Status.TimeOn > 30*time.Minute &&
+			time.Now().Sub(mp.lastCameraTempSave) > 30*time.Minute {
+			log.Printf("saving target temp as %v", frame.Status.TempC)
+			mp.lastCameraTempSave = time.Now()
+			err := ioutil.WriteFile(targetTempFile, []byte(fmt.Sprintf("%f", frame.Status.TempC)), 0644)
+			if err != nil {
+				log.Println(err)
+			}
+		}
+	} else {
+		if frame.Status.TimeOn > maxFFCWarmupDuration {
+			log.Printf("finished camera warmup beacuse warmup time exceeded %v", maxFFCWarmupDuration)
+			mp.finishedWarmup = true
+		} else if frame.Status.TempC >= targetCameraTemp {
+			log.Printf("finished camera warmup beacuse target temp of %v reached or exceeded", targetCameraTemp)
+			mp.finishedWarmup = true
+		} else if frame.Status.TimeOn-frame.Status.LastFFCTime > ffcTriggerPeriod {
+			runFFC()
+		}
+	}
+}
+
+func runFFC() error {
+	log.Println("triggering FFC")
+	conn, err := dbus.SystemBus()
+	if err != nil {
+		return err
```

Comment:
should we log this error? might be good to know if the dbus service went down for some reason